pkg/steps/cluster_claim.go

```diff
@@ -202,13 +212,49 @@ func (s *clusterClaimStep) releaseCluster(ctx context.Context, clusterClaim *hiv
 	return nil
 }
 
-func ClusterClaimStep(as string, clusterClaim *api.ClusterClaim, hiveClient ctrlruntimeclient.WithWatch, client loggingclient.LoggingClient, jobSpec *api.JobSpec, wrapped api.Step) api.Step {
+func (s clusterClaimStep) saveArtifacts(ctx context.Context, namespace, name string) error {
+	logrus.WithField("clusterClaim.Namespace", namespace).WithField("clusterClaim.Name", name).Debug("Saving artifacts.")
+	var errs []error
+	namespaceDir := api.NamespaceDir
+	claim := &hivev1.ClusterClaim{}
+	if err := s.hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: namespace, Name: name}, claim); err != nil {
+		return fmt.Errorf("failed to get the claim %s in namespace %s: %w", name, namespace, err)
+	}
+	data, err := json.MarshalIndent(claim, "", "  ")
+	if err != nil {
+		errs = append(errs, err)
+	}
+	path := filepath.Join(namespaceDir, "claim.json")
+	if err := api.SaveArtifact(s.censor, path, data); err != nil {
+		errs = append(errs, err)
+	}
```

Comment:
If the above `if err != nil` is true, would this logic ever succeed or is it more efficient to skip over it?

Reply:
Skipped the `api.SaveArtifact` call  while err on `json.MarshalIndent`.